src/lib/crypto.c

```diff
@@ -151,6 +169,18 @@ __Z_INLINE int blake_hash(const unsigned char *in, unsigned int inLen,
     return 0;
 }
 
+__Z_INLINE int blake_hash_cid(const unsigned char *in, unsigned int inLen,
+                              unsigned char *out, unsigned int outLen) {
+
+    blake2b_state s;
+    blake2b_init(&s, outLen);
+    blake2b_update(&s, prefix, 6);
```

Comment:
I would replace the 6 with a sizeof(prefix) otherwise changes in prefix length might be ignored.
```suggestion
    blake2b_update(&s, prefix, sizeof(prefix));
```